### PR TITLE
[CI] Skip tests on doc-only changes

### PR DESCRIPTION
**Pull request by greglucas**
## PR Summary

We don't need to waste CI cycles for doc-only changes and having to remember to add `[doc skip]` to commit messages is asking a lot of users. Also removed extra CI conditionals for skipping that are already handled by GitHub: https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs
(except [skip github], but that isn't documented and it should really be [skip actions], so let's just remove all of them).

I tested this on an example PR on my fork to verify: https://github.com/greglucas/matplotlib/pull/21

I only added this to `pull_request:`, we could also add this to `push:`, but it seems fine to run tests on every push to `main` regardless.

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -26,14 +31,7 @@ env:
 jobs:
   test:
     if: |
-      github.event_name == 'workflow_dispatch' ||
-      (
-        github.repository == 'matplotlib/matplotlib' &&
-        !contains(github.event.head_commit.message, '[ci skip]') &&
-        !contains(github.event.head_commit.message, '[skip ci]') &&
-        !contains(github.event.head_commit.message, '[skip github]') &&
-        !contains(github.event.head_commit.message, '[ci doc]')
```
Comment (rcomer) [inline .github/workflows/tests.yml:35]: `[ci doc]` should still be useful if you make a docstring only change.  `[skip github]` is specific to these tests, whereas GitHub’s `[skip actions]` will skip all the GitHub actions.  I don’t know whether that difference is important enough to keep `[skip github]`.

**Comment by rcomer** [comment]
Presumably tests will still run if you edit e.g. both an example and some mpl functionality?  That doesn’t seem obvious from the GitHub docs.

**Comment by greglucas** [comment]
> Presumably tests will still run if you edit e.g. both an example and some mpl functionality? That doesn’t seem obvious from the GitHub docs.

Yes, it should cover that case, with an example here:

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-excluding-paths

> When all the path names match patterns in paths-ignore, the workflow will not run. If any path names do not match patterns in paths-ignore, even if some path names match the patterns, the workflow will run.

**Review by rcomer — APPROVED** [review]

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -12,6 +12,11 @@ on:
   pull_request:
     branches-ignore:
       - v[0-9]+.[0-9]+.[0-9x]+-doc
+    paths-ignore:
+      # No need to test on documentation changes
```
Comment (jklymak) [inline .github/workflows/tests.yml:16]: I find the wording here a bit ambiguous. Maybe "skip running tests if changes are only in the documentation directories"?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -12,6 +12,11 @@ on:
...
+      - 'doc/**'
+      - 'examples/**'
+      - 'tutorials/**'
```
Comment (jklymak) [inline .github/workflows/tests.yml:19]: Need to add plot_types.  Note if we get #25209 in we can just test on "galleries/**"

File: .github/workflows/tests.yml (hunk 2 above)
Comment (greglucas) [reply .github/workflows/tests.yml]: Good suggestion, thanks.

File: .github/workflows/tests.yml (hunk 3 above)
Comment (greglucas) [reply .github/workflows/tests.yml:20]: :+1: We can add more later no problem. I guess an alternative question for the future is whether we really want these at the top-level of the repository versus living under `doc/galleries`, `doc/plot_types` etc... which would then allow just `doc/**` to be ignored.

Comment (greglucas) [reply .github/workflows/tests.yml:20]: Ahh, I see. I have not been following the docs reorganization. That does seem like a nice change!

**Review by jklymak — APPROVED** [review]

**Comment by jklymak** [comment]
I'll merge since there is no point in running CI on a CI change...

**Pull request merged**